model-optimizer/extensions/ops/fakequantize.py

```diff
@@ -63,7 +63,7 @@ def infer(node: Node):
 
         if all([node.in_node(i).has_valid('value') for i in range(5)]):
             x, input_low, input_high, output_low, output_high = \
-                [np.array(np.broadcast_to(node.value, x.value.shape), dtype=np.float32) for node in inputs]
+                [np.array(np.broadcast_to(node.value, x.value.shape), dtype=inputs[0].value.dtype) for node in inputs]
```

Comment:
Does it guarantee that all the intermediate computations that are related to compression transformation will be done in float32 still? I think this is critical to have a precise computation of zero-point and scale while storing them in the target data type.

Reply:
@rkazants you are right. But POT with old IR will regress

Reply:
@AlexKoff88 Now it's guaranteed. But I still think that picking limits of FQ of f16 model from the hardware with f32 compute is wrong. I can provide you with examples where f16 computations gave inaccurate results.